plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotConnectorTest.java

```diff
@@ -0,0 +1,379 @@
+                //.put("pinot.max-rows-per-split-for-segment-queries", String.valueOf(MAX_ROWS_PER_SPLIT_FOR_SEGMENT_QUERIES))
+                //.put("pinot.max-rows-for-broker-queries", String.valueOf(MAX_ROWS_PER_SPLIT_FOR_BROKER_QUERIES))
+                .putAll(additionalPinotProperties())
+                .buildOrThrow();
+    }
+
+    protected Map<String, String> additionalPinotProperties()
+    {
+        if (isGrpcEnabled()) {
+            return ImmutableMap.of("pinot.grpc.enabled", "true");
+        }
+        return ImmutableMap.of();
+    }
+
+    protected boolean isGrpcEnabled()
+    {
+        return true;
+    }
+
+    protected boolean isSecured()
+    {
+        return false;
+    }
+
+    protected String getPinotImageName()
+    {
+        return PINOT_LATEST_IMAGE_NAME;
+    }
+
+    private void createAndPopulateNationAndRegionData(TestingKafka kafka, TestingPinotCluster pinot, DistributedQueryRunner queryRunner)
+            throws Exception
+    {
+        // Create and populate table and topic data
+        String regionTableName = "region";
+        kafka.createTopicWithConfig(2, 1, regionTableName, false);
+        Schema regionSchema = SchemaBuilder.record(regionTableName).fields()
+                // regionkey bigint, name varchar, comment varchar
+                .name("regionkey").type().longType().noDefault()
+                .name("name").type().stringType().noDefault()
+                .name("comment").type().stringType().noDefault()
+                .name("updated_at_seconds").type().longType().noDefault()
+                .endRecord();
+        ImmutableList.Builder<ProducerRecord<String, GenericRecord>> regionRowsBuilder = ImmutableList.builder();
+        MaterializedResult regionRows = queryRunner.execute("SELECT * FROM tpch.tiny.region");
+        for (MaterializedRow row : regionRows.getMaterializedRows()) {
+            regionRowsBuilder.add(new ProducerRecord<>(regionTableName, "key" + row.getField(0), new GenericRecordBuilder(regionSchema)
+                    .set("regionkey", row.getField(0))
+                    .set("name", row.getField(1))
+                    .set("comment", row.getField(2))
+                    .set("updated_at_seconds", initialUpdatedAt.plusMillis(1000).toEpochMilli())
+                    .build()));
+        }
+        kafka.sendMessages(regionRowsBuilder.build().stream(), schemaRegistryAwareProducer(kafka));
+        pinot.createSchema(getClass().getClassLoader().getResourceAsStream("region_schema.json"), regionTableName);
+        pinot.addRealTimeTable(getClass().getClassLoader().getResourceAsStream("region_realtimeSpec.json"), regionTableName);
+
+        String nationTableName = "nation";
+        kafka.createTopicWithConfig(2, 1, nationTableName, false);
+        Schema nationSchema = SchemaBuilder.record(nationTableName).fields()
+                // nationkey BIGINT, name VARCHAR,  VARCHAR, regionkey BIGINT
```

Comment:
This comment looks unhelpful. Same for others.

Reply:
Thanks for catching! Left that over from testing, removed.